StRoot/StFcsWaveformFitMaker/PeakAna.cxx

```diff
@@ -0,0 +1,1336 @@
+void PeakAna::SetWindow(const Int_t start, const Int_t end )
+{ mFoundPeak.SetWindow(start,end); }
+
+void PeakAna::SetWindow(PeakWindow window)
+{ mFoundPeak=window; }
+
+void PeakAna::SetTunnelScale(Double_t value)
+{
+  if( value>=0 ){mTunnelScale = value;}
+}
+void PeakAna::SetTunnelSigma(Double_t value)
+{
+  if( value>0 ){mTunnelSigma = value;}
+}
+void PeakAna::SetTunnelPars(Double_t scale, Double_t sigma)
+{
+  SetTunnelScale(scale);
+  SetTunnelSigma(sigma);
+}
+void PeakAna::SetTunnelThreshold(Double_t value)
+{
+  if( value<=1 ){ mTunnelThreshold=value; }
+}
+
+
+TGraph* PeakAna::ConvertHistToGraph(TH1* hist, UInt_t numavgs)
+{
+  Int_t nbins = hist->GetNbinsX();
+  //Only consider averaging when number of bins to average are greater than or equal to 1 or less than the number of bins
+  if( numavgs<1 || static_cast<Int_t>(numavgs)>nbins ){return 0;}//strictly greater than since GetNbinsX()+1 is overflow
+  TGraph* AvgGr = new TGraph();
+  for( Int_t ibin=1; ibin<=nbins; ibin+=numavgs ){
+    Double_t sum=0;
+    UInt_t counter=0;//In case number of averages is not same as how many sums were performed
+    for(Int_t i=ibin; i<ibin+static_cast<Int_t>(numavgs) && i<=nbins; ++i ){ sum += hist->GetBinContent(i); ++counter; }
+    Double_t avg = sum/static_cast<Double_t>(counter);
+    //std::cout << std::endl<< "|sum:"<<sum << "|avg:"<<avg <<"|val:"<<hist->GetBinContent(ibin) << std::endl;
+    //Set graph's point to center of x range
+    Double_t xlow = hist->GetBinLowEdge(ibin);
+    Double_t xhigh = hist->GetBinLowEdge(ibin+counter);
+    //AvgGr->SetPoint(AvgGr->GetN(),static_cast<Double_t>(ibin)+static_cast<Double_t>(numavgs)*0.5,avg);//This is for knowing which bin
+    AvgGr->SetPoint(AvgGr->GetN(),(xlow+xhigh)/2.0,avg);//This is for knowing the x-value
+  }
+  return AvgGr;
+}
+
+TGraph* PeakAna::ConvertPeaksToGraph(const std::vector<PeakWindow> &Peaks)
+{
+  if( Peaks.size()==0 ){return 0;}
+  TGraph* graph = new TGraph();
+  for( UInt_t ipeak=0; ipeak<Peaks.size(); ++ipeak )
+  {
+    graph->SetPoint( ipeak,Peaks.at(ipeak).mPeakX,Peaks.at(ipeak).MidPoint() );
+  }
+  return graph;
+}
+
+void PeakAna::ConvertPeaksToGraph()
+{
+  TGraph* graph = new TGraph();
+  for( UInt_t ipeak=0; ipeak<mPeaks.size(); ++ipeak )
+  {
+    graph->SetPoint( ipeak,mPeaks.at(ipeak).mPeakX,mPeaks.at(ipeak).MidPoint() );
+  }
+  this->SetData(graph);
+  this->ForceInternal();//new graph replaces old one so needs to be deleted
+  return;
+}
+
+PeakAna* PeakAna::ConvertPeaksToAna(const PeakAna &Ana)
+{
+  TGraph* graph = new TGraph();
+  for( Int_t ipeak=0; ipeak<Ana.NPeaks(); ++ipeak ){
+    graph->SetPoint(ipeak, Ana.GetPeak(ipeak).mPeakX, Ana.GetPeak(ipeak).MidPoint() );
+  }
+  PeakAna* NewAna = new PeakAna(Ana,graph);
+  NewAna->ForceInternal();//new graph so needs to be deleted
+  return NewAna;
+}
+
+PeakAna* PeakAna::MeanFilter( Int_t sizeavgs, bool copy )
+{
+  if( this->GetData()==0 ){return this;}
```

Comment:
You use `this->` a lot for class members. In most of our code it is usually omitted. I suggest you do the same.

Reply:
Removed what I found. I only left the ones relating to virtual methods.